### PR TITLE
Streamline onboarding: Conditionally hide the tax setting

### LINKED ISSUE #2490: Onboarding: Remove Tax Rates Selection UI
Part of https://github.com/woocommerce/google-listings-and-ads/issues/2458

Image

This 2nd step of onboarding includes a section for selecting whether the store uses destination based tax rates. This does not need to be chosen during initial onboarding, so we’ll remove this step and default to destination-based tax ratesduring onboarding.

## Acceptance Criteria

- [ ] The tax rate section at the bottom of step 2 is no longer present on this screen.
- [ ] After onboarding, when you edit a free campaign, the tax rate section should show when the audience includes 'US' (retain current behavior)
- [ ] After onboarding, when you edit a free campaign, the tax rate section should not show when the audience does not include 'US' (retain current behavior)

## Implementation Brief

The component that renders this section of the UI is `FormContent` in `js/src/components/free-listings/setup-free-listings/form-content.js`. Here, there is a `ConditionalSection` component that wraps the `TaxRate` component. The `SetupFreeListings` component that wraps this `FormContent` component is shared between the `SavedSetupStepper` for the onboarding and the `EditFreeCampaign` component. 

Since we want to still show the ability to edit the tax rates for the `EditFreeCampaign`, we will need to add an optional prop like `hideTaxRates` to `FormContent` to optionally control whether those controls are shown. That prop should default to `false`, but if `true` should be used in `FormContent` as part of the logic used to set the value of `shouldDisplayTaxRate` ([ref](https://github.com/woocommerce/google-listings-and-ads/blob/trunk/js/src/components/free-listings/setup-free-listings/form-content.js#L31)).

## Test Coverage

Update the E2E tests in `tests/e2e/specs/setup-mc/step-2-product-listings.test.js` to remove tests that are no longer needed.

## Definition Questions

1. The `handleSubmitClick` callback includes a conditional check for `shouldDisplayTaxRate`. Is it safe to remove this conditional and the hook that this value is used to get this value since we're not longer using it?
2. Any other adaptiveFormContext that needs to be updated?

### PR DESCRIPTION
**Pull request by dsawardekar**
### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/google-listings-and-ads/issues/2490

This PR removes the tax rates UI during the onboarding steps. The UI is preserved for the edit campaign flow.

Note: This PR depends on the work in https://github.com/woocommerce/google-listings-and-ads/pull/2543/files. 

### Screenshots:

Onboarding:
image

Edit Campaign:
image

### Detailed test instructions:

Dependency:
Merge https://github.com/woocommerce/google-listings-and-ads/pull/2543/files into your local branch before testing this feature. Without this you will not be able to continue to Step 3.

1. Start the onboarding steps from a clean slate.
2. You should not see any Tax Rate section
3. You should be able to complete Step 2 and proceed by clicking the Continue button
4. After onboarding, edit the free Campaign 
5. You should see the Tax Rate UI that was removed during onboarding

### Additional details:

Tweak - Conditionally hide the tax setting during onboarding

### Changelog entry

> Update - Hides the tax rate setting during onboarding.

### CODE REVIEW

File: js/src/edit-free-campaign/index.js [hunk 1]
```diff
@@ -208,6 +208,7 @@ const EditFreeCampaign = () => {
 				settings={ settings }
 				onSettingsChange={ updateSettings }
 				shippingRates={ loadedShippingRates }
+				hideTaxRates={ false }
```
Comment (joemcgill) [inline js/src/edit-free-campaign/index.js]: Since this is using the default, this prop can be omitted.
```suggestion
```

**Review by joemcgill — CHANGES_REQUESTED** [review]
I've left one suggestion inline. Otherwise, the main thing that needs to be addressed is that I see an E2E test that's failing related to the validation logic that happens in the `SetupFreeListings` component:

https://github.com/woocommerce/google-listings-and-ads/blob/7acafdf4dbe3c6fa0edd778a483d5a68ac32464d/js/src/components/free-listings/setup-free-listings/index.js#L104-L114

The `checkErrors()` function includes this logic to validate the `tax_rate` value for 'US' based stores:

https://github.com/woocommerce/google-listings-and-ads/blob/7acafdf4dbe3c6fa0edd778a483d5a68ac32464d/js/src/components/free-listings/configure-product-listings/checkErrors.js#L109-L120

We're setting this default for tax rates via the API in #2491, so we need to update this so that the validation for `tax_rates` only runs if `hideTaxRates` is `false`.

To verify this locally, set your store location to 'United States' and try completing this form. You will be able to click the "Continue" button, but nothing will happen and no errors will be shown.

**Comment by dsawardekar** [comment]
@joemcgill I've addressed the PR feedback, can you take another look? thx

**Review by ankitrox — APPROVED** [review]
LGTM 💯

**Review by joemcgill — APPROVED** [review]
I traced down the E2E test failure that I was seeing since I couldn't reproduce it anymore now that [57ecc5d](https://github.com/woocommerce/google-listings-and-ads/pull/2542/commits/57ecc5dbb81d42403ab01113325aad25b2babb04) got merged into this branch. Turns out the default settings get mocked in these tests and needed to be updated in [here](https://github.com/woocommerce/google-listings-and-ads/pull/2542/commits/c2c324212806bdd0d35003c37a2fd651c57efa46).

I went ahead and cleaned up a few more test methods that are no longer used after these changes—something to keep an eye going forward when deleting tests.

**Comment by ankitguptaindia** [comment]
**QA/Test Report-** ✅ 
------------

**Testing Environment -** 

<Details>

* WordPress: 6.6.1
* Theme active on store: Storefront Version: 4.6.0
* WooCommerce - Version 9.2.3 
* PHP: 8.3
* Web Server: Nginx
* Browser: Chrome - Version 127
* OS: macOS Sonoma 14.6

</Details>



**Test Results -**

- **Positive Test:**

    - The tax selection UI does not appear during onboarding but does appear on the edit campaign screen. ✅
    - The tax selection UI appears when the audience location is set to the US. ✅

- **Negative Test:** The tax selection UI should not appear when the audience location is set to a country outside the US. ✅

- **Default Selection of Tax UI:** "My store uses destination-based tax rates." in the edit campaign screen. ✅ 


**Functional Demo /  Screencast -**

https://github.com/user-attachments/assets/d7b20ff3-4e75-4231-8ccd-21317500bbe7




**Plugin file with this PR related code to test** [google-listings-and-ads.zip](https://github.com/user-attachments/files/16781259/google-listings-and-ads.zip)

**Review by ankitguptaindia — APPROVED** [review]

File: js/src/components/free-listings/setup-free-listings/form-content.js [hunk 2]
```diff
@@ -22,13 +22,16 @@ import ConditionalSection from '.~/components/conditional-section';
  *
  * @param {Object} props React props.
  * @param {string} [props.submitLabel="Complete setup"] Submit button label.
+ * @param {boolean} [props.hideTaxRates=false] Whether to hide tax rate section.
  */
 const FormContent = ( {
 	submitLabel = __( 'Complete setup', 'google-listings-and-ads' ),
+	hideTaxRates = false,
```
Comment (eason9487) [inline js/src/components/free-listings/setup-free-listings/form-content.js]: The default value of `hideTaxRates` could be removed as the `SetupFreeListings` component already set it and this component is considered as a sub-component of `SetupFreeListings`.

File: js/src/components/free-listings/setup-free-listings/index.js [hunk 3]
```diff
@@ -77,6 +77,7 @@ const getSettings = ( values ) => {
  * @param {() => void} [props.onContinue] Callback called once continue button is clicked. Could be async. While it's being resolved the form would turn into a saving state.
  * @param {string} [props.submitLabel] Submit button label, to be forwarded to `FormContent`.
  * @param {JSX.Element} props.headerTitle Title in the header block of this setup.
+ * @param {boolean} [props.hideTaxRates] Whether to hide tax rate section, to be forwarded to `FormContent`.
```
Comment (eason9487) [inline js/src/components/free-listings/setup-free-listings/index.js]: It would be better to declare the default value as well.

**Review by eason9487 — COMMENTED** [review]
Thanks for the change! It works well.

Could you help with removing an unused method?

https://github.com/woocommerce/google-listings-and-ads/blob/4870093d73814c38c835c9498017dda2f2494849/tests/e2e/utils/pages/setup-mc/step-2-product-listings.js#L138-L148

**Comment by joemcgill** [comment]
@eason9487 I think this is ready for a second review.

File: js/src/components/free-listings/setup-free-listings/form-content.js [hunk 4]
```diff
@@ -22,13 +22,16 @@ import ConditionalSection from '.~/components/conditional-section';
  *
  * @param {Object} props React props.
  * @param {string} [props.submitLabel="Complete setup"] Submit button label.
+ * @param {boolean} [props.hideTaxRates=false] Whether to hide tax rate section.
```
Comment (eason9487) [inline js/src/components/free-listings/setup-free-listings/form-content.js]: This JSDoc needs to be updated as well.

```suggestion
 * @param {boolean} props.hideTaxRates Whether to hide tax rate section.
```

**Review by eason9487 — APPROVED** [review]
Thanks for the update!

Although there are JSDoc and code conflict need to be adjusted, I believe it should not need another round of code reviews after resolving them.

**Comment by dsawardekar** [comment]
@joemcgill I have updated the jsdoc and fixed the merge conflict. Assigning to you to take another look.

**Comment by asvinb** [comment]
Change LGTM @dsawardekar . As per @eason9487 's comment, I'am reassigning the ticket to @fblascogarma for UAT.

**Comment by joemcgill** [comment]
Since this has been approved, I'm merging into the feature branch.

**Pull request merged**